src/Illuminate/Queue/Worker.php

```diff
@@ -291,6 +291,8 @@ public function process($connectionName, $job, WorkerOptions $options)
             $job->fire();
 
             $this->raiseAfterJobEvent($connectionName, $job);
+
+            $job->finished();
```

Comment:
Should the `finished()` before called before or after the event?

If you want to run "job specific" cleanup code - you would realistically do that in the new `finished()` function.

And if you want to run "all job" cleanup code after each job - you would realistically want to run that last - after the job does anything special it feels it needs to do?